runtime/doc/lua.txt

```diff
@@ -1581,4 +1581,45 @@ uri_to_fname({uri})                                       *vim.uri_to_fname()*
                 Return: ~
                     Filename
 
+==============================================================================
+Lua module: xdiff                                                  *lua-xdiff*
+
+xdl_diff({a}, {b}, {opts})                                    *vim.xdl_diff()*
+                Run diff on strings {a} and {b}.
+
+                Parameters: ~
+                    {a}      First string to compare
+                    {b}      Second string to compare
+                    {opts}   Optional parameters:
+                             • `hunk_func`: Lua callback invoked for each hunk
```

Comment:
implementing iteration as callbacks in a loop is not considered efficient in luajit. It would be good with a `hunk_list` option which does the `hunk_func = function(sa, ca, sb, cb)
          exp[#exp+1] = {sa, ca, sb, cb}
        end` thing everyone is gonna do anyway (but without the overhead of re-entering the interpreter each item)

Reply:
Great idea. Done